tests/test_filtering.py

```diff
@@ -87,6 +87,97 @@ def test_sigmag_filtered_indices_three_outliers(self):
         for i in range(29):
             valid = (i != 13 and i != 14 and i != 27)
             self.assertEqual(i in inds, valid)
+            
+    def test_kalman_filtered_indices(self):
+        # With everything the same, nothing should be filtered.
+        psi_values = [[1.0 for _ in range(20)] for _ in range(20)]
+        phi_values = [[1.0 for _ in range(20)] for _ in range(20)]
+        
+        inds = kalman_filtered_indices(psi_values, phi_values)
+        
+        self.assertEqual(len(inds), 20)
+        for i in inds:
+            self.assertFalse(i[1] == [-1])
+            self.assertEqual(i[2], 4.47213595499958)
+            
+    def test_kalman_filtered_indices_zero_phi(self):
+        # make sure kalman filtering properly handles when a phi
+        # value is 0. The last value should be masked and filtered out.
+        psi_values = [[1.0 for _ in range(20)] for _ in range(21)]
+        phi_values = [[1.0 for _ in range(20)] for _ in range(20)]
+        phi_values.append([0.0 for _ in range(20)])
+        
+        inds = kalman_filtered_indices(psi_values, phi_values)
+        
+        self.assertEqual(len(inds), 21)
+        for i in inds[:-1]:
+            self.assertFalse(i[1] == [-1])
+            self.assertEqual(i[2], 4.47213595499958)
+        self.assertEqual(inds[-1][2], 0.0)
+        
+    def test_kalman_filtered_indices_negative_phi(self):
+        # make sure kalman filtering properly handles when a phi
+        # value is less than -999.0. The last value should be 
+        # xmasked and filtered out.
+        psi_values = [[1.0 for _ in range(20)] for _ in range(21)]
+        phi_values = [[1.0 for _ in range(20)] for _ in range(20)]
+        phi_values.append([-999.1 for _ in range(20)])
+        
+        inds = kalman_filtered_indices(psi_values, phi_values)
+        
+        self.assertEqual(len(inds), 21)
+        for i in inds[:-1]:
+            self.assertFalse(i[1] == [-1])
+            self.assertEqual(i[2], 4.47213595499958)
+        self.assertEqual(inds[-1][2], 0.0)
+        
+    def test_kalman_filtered_indices_negative_flux(self):
+        # make sure kalman filtering filters out all indices with
+        # a negative flux value.
+        psi_values = [[-1.0 for _ in range(20)] for _ in range(20)]
+        phi_values = [[1.0 for _ in range(20)] for _ in range(20)]
+        phi_values.append([-999.1 for _ in range(20)])
+        
+        inds = kalman_filtered_indices(psi_values, phi_values)
+        
+        self.assertEqual(len(inds), 20)
+        for i in inds[:-1]:
+            self.assertTrue(i[1] == [-1])
+            self.assertEqual(i[2], 0.0)
+            
+    def test_kalman_filtered_indices_bright_first_obs(self):
```

Comment:
Can you also add a test where the kalman filter rejects one of the observations in the middle? Maybe something like:
```
psi_values = [[10., 10., 0.001, 10., 10.]]
phi_values = [[1., 1., 1., 1., 1]]
```